core/state_transition.go

```diff
@@ -336,7 +336,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		// Record deposits as using all their gas (matches the gas pool)
 		// System Transactions are special & are not recorded as using any gas (anywhere)
 		gasUsed := st.msg.Gas()
-		if st.msg.IsSystemTx() {
+		if st.msg.IsSystemTx() && !st.evm.ChainConfig().IsRegolith(st.evm.Context.Time) {
```

Comment:
Needs a comment update about the regolith tweak.

Reply:
Added comments - let me know if they fit what you expect.